refactor(feature:share-account): Complete CMP Migration

HekmatullahAmin opened the pull request:
Fixes - [Jira-#188](https://mifosforge.jira.com/browse/MM-188)

Didn't create a Jira ticket, click [here](https://mifosforge.jira.com/jira/software/c/projects/MM/issues/) to create new.

Please Add Screenshots If there are any UI changes.

| Before                                     | After                                  |
|--------------------------------------------|----------------------------------------|
|                                            |                                        |


Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Run the static analysis check `./gradlew check` or `ci-prepush.sh` to make sure you didn't break anything

- [ ] If you have multiple commits please combine them into one commit by squashing them.

HekmatullahAmin committed:
refactor(feature:share-account): Address remaining issues from CMP migration

niyajali commented on feature/share-account/src/commonMain/kotlin/org/mifos/mobile/feature/shareaccount/utils/FilterUtil.kt:
```diff
@@ -14,34 +14,53 @@ import mifos_mobile.feature.share_account.generated.resources.feature_account_ac
+ * @property label The string resource representing the filter name.
+ * @property matchCondition A lambda function that checks if a [ShareAccount] meets the filter condition.
+ */
+enum class FilterUtil(val label: StringResource, val matchCondition: (ShareAccount) -> Boolean) {
```
Format these enum class

niyajali commented on feature/share-account/src/commonMain/kotlin/org/mifos/mobile/feature/shareaccount/screen/ShareAccountScreen.kt:
```diff
@@ -10,73 +10,99 @@
+            is AccountState.Success -> {
+                val shareAccounts = state.shareAccounts
+                if (shareAccounts.isNullOrEmpty()) {
+                    EmptyDataView(
```
Create one more state as AccountState.Empty for empty screen..

HekmatullahAmin committed:
apply suggested changes

HekmatullahAmin commented:
Hi Mr. @therajanmaurya , @niyajali 

I hope you’re doing well. I’ve applied the suggestions:

- Created another state to display empty data.
- Formatted some lines for better readability.
- Moved the filter logic inside the loadAccounts function within the viewModel, eliminating the need to call it again inside the Screen.
- Additionally, I updated the Share accounts item label so that its status is now dynamically set based on the Status property of the data class ShareAccount, instead of using the predefined string "pending."

Could you please review these updates? Kindly approve and merge at your convenience.
